hamilflow/models/d0/complex_harmonic_oscillator.py

```diff
@@ -0,0 +1,87 @@
+from dataclasses import asdict, dataclass, field
+from functools import cached_property
+from typing import Mapping
+
+import numpy as np
+from numpy.typing import ArrayLike
+from pydantic import BaseModel, Field, computed_field
+
+
+class ComplexHarmonicOscillatorSystem(BaseModel):
```

Comment:
Out of curiosity, why not reuse the HarmonicOscillatorSystem class? Seems like you've extend the HarmonicOscillatorSystem to support complex cases.

Reply:
https://github.com/kausalflow/hamilflow/pull/58/commits/bc73c3320a2a8c89aae3780cb2ddafb4a305a992